### ISSUE TITLE
App keeps crashing after clicking download and bookmark the video. 

### ISSUE THREAD

**Issue by okan35** [ram-on/SkyTube#225]
A bug that can be reproduced in following way; If you click "download" and then bookmark the video, the apps crashes and keeps stopping if you try to open it again. It opens only when you clean data of the app, not even cache.

**Comment by atomjack** [comment]
@okan35 I'm not able to reproduce this. Can you elaborate on the steps you took? A couple questions:

1) Were you running the latest commit from master? I tried it with db0aa6f9122bc2a7a7d3f96492a76dd478ad2853

2) Which tab were you on when you clicked download? 

3) Which tab were you on when you clicked bookmark? Was it from the tab you clicked download, or did you move to the downloads tab first?

4) After you clicked download, how long did you wait until you clicked bookmark? Was it right away, while the download was in progress (when the notification appeared), or was it after it was complete?

I tried a few different variations, but it always bookmarked it just fine. I even tried clearing the app data and trying it fresh and it still didn't crash.

**Comment by okan35** [comment]
@atomjack I am getting this error on this commit. https://github.com/okan35/SkyTube/commit/593fe2e48c504a9f6b89fb042c2b39c7ea0f447e

I just click on the options of a video and bookmark it I get the following. It shows in the for loop that I implemented to filter channels that are going to be got but this happens when I bookmark and I did nothing about bookmark feature. I will work on this.

`01-12 18:44:37.831 24239-24239/free.rm.skytube.extra E/AndroidRuntime: FATAL EXCEPTION: main
                                                                       Process: free.rm.skytube.extra, PID: 24239
                                                                       java.lang.NullPointerException: Attempt to invoke interface method 'java.util.Iterator java.util.List.iterator()' on a null object reference
                                                                           at free.rm.skytube.businessobjects.YouTube.Tasks.GetYouTubeVideosTask.onPostExecute(GetYouTubeVideosTask.java:132)
                                                                           at free.rm.skytube.businessobjects.YouTube.Tasks.GetYouTubeVideosTask.onPostExecute(GetYouTubeVideosTask.java:38)
                                                                           at android.os.AsyncTask.finish(AsyncTask.java:695)
                                                                           at android.os.AsyncTask.-wrap1(Unknown Source:0)
                                                                           at android.os.AsyncTask$InternalHandler.handleMessage(AsyncTask.java:712)
                                                                           at android.os.Handler.dispatchMessage(Handler.java:105)
                                                                           at android.os.Looper.loop(Looper.java:251)
                                                                           at android.app.ActivityThread.main(ActivityThread.java:6563)
                                                                           at java.lang.reflect.Method.invoke(Native Method)
                                                                           at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
                                                                           at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)`

**Comment by atomjack** [comment]
Ok, well first off you should have mentioned that you were getting the error on changes you had made in your own fork. Second, you shouldn't be adding commits to master (in your fork), you should be adding commits to a separate branch. Once you are done you push that branch up to github (to your fork) and then can do a Pull Request to have that branch merged into ram-on/master.

The problem appears to be that sometimes `onPostExecute` can be called, passing a null value for `videosList`. Before your changes, all it was doing with that ArrayList was appending it to the `videoGridAdapter` - if you follow that `videoGridAdapter.appendList` method, you'll see it's checking to see if the passed list is null or empty - if it is either, it just doesn't do anything. I'd suggest you make sure that `videosList` is not null before trying to iterate over it. What will end up happening in this case is it will call `videoGridAdapter.appendList` and pass an empty (initialized by with 0 elements) ArrayList.

Lastly, in your `BlockedChannelsDb.getBlockedChannelsList` method, you're using `getWritableDatabase` to get a reference to the database, but all you're doing in that method is reading from the database. You should use `getReadableDatabase` instead (and you don't need `this.`). And instead of doing `SELECT * FROM <table>`, you should only be querying the columns you actually need (in this case, just the channel id). Just a suggestion.

**Comment by okan35** [comment]
I should have checked better but I fixed the issue and  so far it seems fine and I haven't got any problems. The reason that I made changes to master was I just could not make changes to branch so I just did it to master.

I will also work on your suggestions I believe I am still a beginner so I can make mistakes. Thanks for your suggestions I will definitely work on them I just want this feature to be available in this app as it is very important feature at least for me.
